packages/gradle/src/migrations/19-4-1/change-regex-test-production.ts

```diff
@@ -0,0 +1,19 @@
+import { Tree, readNxJson } from '@nx/devkit';
+import { hasGradlePlugin } from '../../utils/has-gradle-plugin';
+import { updateNxJsonConfiguration } from '../../generators/init/init';
+
+// This function changes !{projectRoot}/test/**/* in nx.json for production to !{projectRoot}/src/test/**/*
+export default function update(tree: Tree) {
+  if (!hasGradlePlugin(tree)) {
+    return;
+  }
+  if (tree.exists('nx.json')) {
+    const nxJson = readNxJson(tree);
+    const production = nxJson?.namedInputs?.production;
+    if (production?.includes('!{projectRoot}/test/**/*')) {
+      production.splice(production.indexOf('!{projectRoot}/test/**/*'), 1);
+      tree.write('nx.json', JSON.stringify(nxJson, null, 2));
+    }
+    updateNxJsonConfiguration(tree);
```

Comment:
Don't reuse this function.

1. Do not update `production` namedInput if it does not exist.
2. Replace the incorrect one with the correct one when it does exist